nvflare/private/fed/client/fed_client_base.py

```diff
@@ -336,63 +326,38 @@ def quit_remote(self, project_name, fl_ctx: FLContext):
         """
         return self.communicator.quit_remote(self.servers, project_name, self.token, self.ssid, fl_ctx)
 
+    def _get_project_name(self):
+        """Get name of the project that the site is part of.
+
+        Returns:
+
+        """
+        s = tuple(self.servers)  # self.servers is a dict of project_name => server config
+        return s[0]
+
     def heartbeat(self, interval):
         """Sends a heartbeat from the client to the server."""
-        pool = None
-        try:
-            pool = ThreadPool(len(self.servers))
-            return pool.map(partial(self.send_heartbeat, interval=interval), tuple(self.servers))
-        finally:
-            if pool:
-                pool.terminate()
+        return self.send_heartbeat(self._get_project_name(), interval)
 
     def pull_task(self, fl_ctx: FLContext):
         """Fetch remote models and update the local client's session."""
-        pool = None
-        try:
-            pool = ThreadPool(len(self.servers))
-            self.remote_tasks = pool.map(partial(self.fetch_execute_task, fl_ctx=fl_ctx), tuple(self.servers))
-            pull_success, task_name, shareable = _check_progress(self.remote_tasks)
-            # TODO: if some of the servers failed
-            return pull_success, task_name, shareable
-        finally:
-            if pool:
-                pool.terminate()
+        result = self.fetch_execute_task(self._get_project_name(), fl_ctx)
+        if result:
+            shareable = result.payload
+            return True, shareable.get_header(ServerCommandKey.TASK_NAME), shareable
+        else:
+            return False, None, None
 
     def push_results(self, shareable: Shareable, fl_ctx: FLContext):
         """Push the local model to multiple servers."""
-        pool = None
-        try:
-            pool = ThreadPool(len(self.servers))
-            return pool.map(partial(self.push_execute_result, shareable=shareable, fl_ctx=fl_ctx), tuple(self.servers))
-        finally:
-            if pool:
-                pool.terminate()
+        return self.push_execute_result(self._get_project_name(), shareable, fl_ctx)
```

Comment:
The original way "theoretically" can map this to multiple servers.

Although in reality we never support multiple servers.
I like the change to make things cleaner.
If later we need to support multiple servers/services we need to look at the whole system and do it.

Reply:
Yes, but that original way was not well thought. Even if we need to support multiple servers at the same time, the simple pool map won't work. And I doubt we'll ever need that.